AsyncDisplayKit/TextKit/ASTextKitRenderer+Positioning.mm

```diff
@@ -333,22 +333,30 @@ - (void)enumerateTextIndexesAtPosition:(CGPoint)externalPosition usingBlock:(as_
 - (CGRect)trailingRect
 {
   __block CGRect trailingRect = CGRectNull;
+  __block CGSize calculatedSize = CGSizeZero;
+  __block NSRange textRange = NSMakeRange(0, 0);
   [self.context performBlockWithLockedTextKitComponents:^(NSLayoutManager *layoutManager, NSTextStorage *textStorage, NSTextContainer *textContainer) {
     CGSize calculatedSize = textContainer.size;
     // If have an empty string, then our whole bounds constitute trailing space.
     if ([textStorage length] == 0) {
       trailingRect = CGRectMake(0, 0, calculatedSize.width, calculatedSize.height);
       return;
     }
-
-    // Take everything after our final character as trailing space.
-    NSArray *finalRects = [self rectsForTextRange:NSMakeRange([textStorage length] - 1, 1) measureOption:ASTextKitRendererMeasureOptionLineHeight];
-    CGRect finalGlyphRect = [[finalRects lastObject] CGRectValue];
-    CGPoint origin = CGPointMake(CGRectGetMaxX(finalGlyphRect), CGRectGetMinY(finalGlyphRect));
-    CGSize size = CGSizeMake(calculatedSize.width - origin.x, calculatedSize.height - origin.y);
-    trailingRect = (CGRect){origin, size};
+      
+    textRange = NSMakeRange([textStorage length] - 1, 1);
```

Comment:
Now that I think about it more, I don't think it's safe to unlock between getting the textRange and calling rectsForTextRange:measureOption: